api/src/handlers/transactions_api.rs

```diff
@@ -121,15 +145,27 @@ impl Handler for TxHashSetHandler {
 		let params = QueryParams::from(req.uri().query());
 		let last_n = parse_param_no_err!(params, "n", 10);
 		let start_index = parse_param_no_err!(params, "start_index", 1);
+		let end_index = match parse_param_no_err!(params, "end_index", 0) {
+			0 => None,
+			i => Some(i),
+		};
 		let max = parse_param_no_err!(params, "max", 100);
 		let id = parse_param_no_err!(params, "id", "".to_owned());
+		let start_height = parse_param_no_err!(params, "start_height", 1);
+		let end_height = match parse_param_no_err!(params, "end_height", 0) {
+			0 => None,
+			h => Some(h),
+		};
 
 		match right_path_element!(req) {
 			"roots" => result_to_response(self.get_roots()),
 			"lastoutputs" => result_to_response(self.get_last_n_output(last_n)),
 			"lastrangeproofs" => result_to_response(self.get_last_n_rangeproof(last_n)),
 			"lastkernels" => result_to_response(self.get_last_n_kernel(last_n)),
-			"outputs" => result_to_response(self.outputs(start_index, max)),
+			"outputs" => result_to_response(self.outputs(start_index, end_index, max)),
+			"heightstopmmr" => result_to_response(
```

Comment:
Subjective comment but "heightstopmmr" is a confusing name... "height stop mmr"?

Also not convinced we want to leak implementation details like this?
Seems like (at least intuitively) that you'd want to get outputs based on height directly, rather than retrieving the intermediate mmr positions.

Once we start doing things based on positions we inevitably go down the path of "is this 1-indexed or 0-index" and "is this leaf insertion indices or actual positions in the MMR"? And it is potientially confusing here because we are looking for leaves, the outputs themselves but via mmr positions, and presumably 1-indexed.

I'd much rather expose this via the api based on block height and keep the MMR implementation hidden behind the api.